### ISSUE TITLE
Scaling metric creates unexpected diagnostic results

### ISSUE THREAD

**Issue by yuzeh** [keras-team/keras#10425]
Hi,

I'm training a model for a regression problem, using `mean_absolute_error` as the loss function.

In addition, I'm reporting a metric that I've created called `m70`, which is exactly the MAE, but multiplied by 70. Here is the implementation:

```python
from keras.losses import mean_absolute_error

def m70(y_true, y_pred):
  return 70 * mean_absolute_error(y_true, y_pred)
```

As the model is training, both numbers are being reported. However, they don't match, i.e. the value for `m70` is not exactly 70x the value for `mean_absolute_error`:

```
4833/4833 [==============================] - 154s - loss: 0.9524 - m70: 14.2387 - val_loss: 0.3981 - val_m70: 2.3467
Epoch 2/50
4833/4833 [==============================] - 150s - loss: 0.3722 - m70: 11.4197 - val_loss: 0.1585 - val_m70: 3.5608
Epoch 3/50
4833/4833 [==============================] - 150s - loss: 0.2093 - m70: 9.8485 - val_loss: 0.1041 - val_m70: 4.7139
Epoch 4/50
4833/4833 [==============================] - 150s - loss: 0.1407 - m70: 8.1068 - val_loss: 0.1429 - val_m70: 8.9124
Epoch 5/50
4833/4833 [==============================] - 150s - loss: 0.1154 - m70: 7.1234 - val_loss: 0.0673 - val_m70: 3.7899
Epoch 6/50
4833/4833 [==============================] - 149s - loss: 0.0944 - m70: 5.6597 - val_loss: 0.1121 - val_m70: 7.2456
```

notice:
```
14.2387 / 0.9524 = 14.950336
11.4197 / 0.3722 = 30.6816228
9.8485 / 0.2093 = 47.0544673
8.1068 / 0.1407 = 57.6176262
7.1234 / 0.1154 = 61.7279029
5.6597 / 0.0944 = 59.9544492
```

I'd expect all of the values above to be 70. Why is this not the case? The only reason I can imagine is that `loss` and `m70` are not evaluated at the same time on the same exact minibatch of data.

FWIW, I'm using:
```
In [1]: import keras
Using TensorFlow backend.
keras.__vers
In [2]: keras.__version__
Out[2]: '2.0.6'

In [3]: import tensorflow

In [4]: tensorflow.__version__
Out[4]: '1.2.1'
```

**Comment by brge17** [comment]
I'm unable to recreate your issue with a ``Dockerfile`` with those exact versions:
```
 6016/60000 [==>...........................] - ETA: 77s - loss: 0.1726 - m70: 12.0841

def m70(y_true, y_pred):
    return 70* keras.losses.mean_absolute_error(y_true, y_pred)

model.compile(loss="mae",
              optimizer=keras.optimizers.Adadelta(),
              metrics=[m70])
```

**Comment by yuzeh** [comment]
@brge17 thanks for looking into this.

The plot thickens... I added MAE to the metrics list in addition to m70, and the metric MAE differs from the loss function, but is consistent with m70:

```
2044/2044 [==============================] - 14s - loss: 0.5226 - m70: 26.1738 - mean_absolute_error: 0.3739 - val_loss: 0.3913 - val_m70: 16.9879 - val_mean_absolute_error: 0.2427
Epoch 2/100
2044/2044 [==============================] - 7s - loss: 0.4822 - m70: 23.3594 - mean_absolute_error: 0.3337 - val_loss: 0.3787 - val_m70: 16.1255 - val_mean_absolute_error: 0.2304
Epoch 3/100
2044/2044 [==============================] - 7s - loss: 0.4468 - m70: 20.8952 - mean_absolute_error: 0.2985 - val_loss: 0.3351 - val_m70: 13.0937 - val_mean_absolute_error: 0.1871
Epoch 4/100
2044/2044 [==============================] - 7s - loss: 0.4263 - m70: 19.4842 - mean_absolute_error: 0.2783 - val_loss: 0.2953 - val_m70: 10.3344 - val_mean_absolute_error: 0.1476
Epoch 5/100
2044/2044 [==============================] - 7s - loss: 0.4023 - m70: 17.8324 - mean_absolute_error: 0.2547 - val_loss: 0.2503 - val_m70: 7.2202 - val_mean_absolute_error: 0.1031
Epoch 6/100
2044/2044 [==============================] - 7s - loss: 0.3809 - m70: 16.3702 - mean_absolute_error: 0.2339 - val_loss: 0.2124 - val_m70: 4.6091 - val_mean_absolute_error: 0.0658
```

I triple checked to make sure the loss function used is `mean_absolute_error` (and not something else like `mean_squared_error`).

I've independently verified that there is no issue when I run on other optimization problems, so my suspicion is that this particular problem is ill-conditioned.

**Comment by yuzeh** [comment]
I figured it out! The "loss" reported by Keras includes any regularization penalities, whereas the metrics do not.
